x-pack/test/security_solution_endpoint/apps/endpoint/resolver.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['common', 'timePicker', 'hosts', 'settings']);
+  const testSubjects = getService('testSubjects');
+  const esArchiver = getService('esArchiver');
+
+  describe('Endpoint Alert Resolver', function () {
+    this.tags(['ciGroup7']);
+
+    before(async () => {
+      await esArchiver.load('endpoint/resolver_tree/api_feature', { useCreate: true });
+      await pageObjects.hosts.navigateToSecurityHostsPage();
+      await pageObjects.common.sleep(4000);
```

Comment:
I'm concerned about this `sleep` call. As far as I can tell you are waiting 4 seconds for the page to load before interacting with it. Here are my reasons for concern:

* This PR will cause CI to pause for 4 seconds before each of these tests. There are 10 tests. If we merge this then from now on every PR will take 40 seconds longer to be validated by CI.
* If the CI server is running slowly, a 4 second wait might not be enough time for your conditions to be met. In that case some of these tests might fail sporadically and for reasons that aren't related to a change in the app's behavior.
* The code defined in the common page objects waits until elements are present before interacting with them by using the [`find` service's `exists` function.](https://github.com/elastic/kibana/blob/633968e0536ab899ddd1c7837ee934a51ebeea58/test/functional/services/common/find.ts#L282)

If you use the `find` service to wait until elements are ready, your code can continue running as soon as the elements are available. In normal situations that means that we won't have to wait 4 seconds for each test. Another advantage: the `find` service's timeouts can be configured by the maintainers of the test framework.

Reply:
removed all sleeps